### PR TITLE
Enable swapping in custom layout mode for "Picture in Picture" style layout

### PR DESCRIPTION
**Pull request by djrobx**
This PR improves the custom layout handling to support a "picture-in-picture" style method of managing the two screens.   

Screen Shot 2022-09-11 at 6 23 27 PM

Previously you could configure this sort of layout, but the swap option was non-functional.   Now you can swap the screens with F9:

Screen Shot 2022-09-11 at 6 23 43 PM

Also adds an opacity setting for the second layer, so the second screen overlay can be semi-transparent. 

Some minor adjustments to also fix the Settings::Apply function to only reset display settings if specified so the input controllers and sound don't get reset with every swap request, as that was causing a loss of button states every time the screens were swapped.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6127)

### CODE REVIEW

**Comment by GPUCode** [comment]
Please run clang format by making the clang-format target (If you cmake run `cmake --build . --target clang-format`)

**Comment by djrobx** [comment]
> Please run clang format by making the clang-format target (If you used cmake run `cmake --build . --target clang-format`)

This doesn't work on Windows/MSVC, unfortunately.

File: src/citra_qt/main.cpp [hunk 1]
```diff
@@ -1708,7 +1708,7 @@ void GMainWindow::ChangeScreenLayout() {
     }
 
     Settings::values.layout_option = new_layout;
-    Settings::Apply();
+    Settings::Apply(true);
```
Comment (SachinVin) [inline src/citra_qt/main.cpp:1711]: You might be able to get away with just calling ` VideoCore::g_renderer->UpdateCurrentFramebufferLayout();`
needs a null check as well.

File: src/video_core/renderer_opengl/renderer_opengl.h [hunk 2]
```diff
@@ -89,6 +89,13 @@ class RendererOpenGL : public RendererBase {
     void ConfigureFramebufferTexture(TextureInfo& texture,
                                      const GPU::Regs::FramebufferConfig& framebuffer);
     void DrawScreens(const Layout::FramebufferLayout& layout, bool flipped);
+    void ResetSecondLayerOpacity();
+    void ApplySecondLayerOpacity();
+    void DrawBottomScreen(const Layout::FramebufferLayout& layout,
+                          const Common::Rectangle<u32>& bottom_screen,
+                          const bool& stereo_single_screen);
```
Comment (SachinVin) [inline src/video_core/renderer_opengl/renderer_opengl.h:96]:
```suggestion
                          bool stereo_single_screen);
```
ditto for the other function and cpp files as well

**Comment by Daisouji** [comment]
rebase

**Comment by FearlessTobi** [comment]
Superseded by https://github.com/citra-emu/citra/pull/6247.

**Pull request closed without merging**